client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx

```diff
@@ -16,15 +19,33 @@ const LaunchpadSitePreview = ( {
 } ) => {
 	const translate = useTranslate();
 	const { globalStylesInUse, shouldLimitGlobalStyles } = usePremiumGlobalStyles();
+	const site = useSite();
 
-	const previewUrl = siteSlug ? 'https://' + siteSlug : null;
+	let previewUrl = siteSlug ? 'https://' + siteSlug : null;
 	const devicesToShow: Device[] = [ DEVICE_TYPES.COMPUTER, DEVICE_TYPES.PHONE ];
 	let defaultDevice = flow === NEWSLETTER_FLOW ? DEVICE_TYPES.COMPUTER : DEVICE_TYPES.PHONE;
-	const isVideoPressFlow = VIDEOPRESS_FLOW === flow;
+	let loadingMessage = translate( '{{strong}}One moment, please…{{/strong}} loading your site.', {
+		components: { strong: <strong /> },
+	} );
 
-	if ( isVideoPressFlow ) {
+	if ( isVideoPressFlow( flow ) ) {
 		const windowWidth = window.innerWidth;
 		defaultDevice = windowWidth >= 1000 ? DEVICE_TYPES.COMPUTER : DEVICE_TYPES.PHONE;
+		const productSlug = site?.plan?.product_slug;
+		const isVideoPressFlowWithUnsupportedPlan = ! planHasFeature(
+			productSlug as string,
+			FEATURE_VIDEO_UPLOADS
+		);
+
+		if ( isVideoPressFlowWithUnsupportedPlan ) {
+			previewUrl = null;
+			loadingMessage = translate(
+				'{{strong}}Site preview not available.{{/strong}} Plan purchase is required.',
```

Comment:
I've seen other flows avoid `purchase` and use something like `upgrade` instead. Maybe `Plan upgrade is required?`